src/gui/widgets/paint_on_template_settings_page.cpp

```diff
@@ -0,0 +1,246 @@
+/*
+ *    Copyright 2020 Libor Pecháček
+ *
+ *    This file is part of OpenOrienteering.
+ *
+ *    OpenOrienteering is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    OpenOrienteering is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with OpenOrienteering.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <algorithm>
+#include <vector>
+
+#include <QAbstractItemModel>
+#include <QAbstractItemView>
+#include <QBrush>
+#include <QColor>
+#include <QColorDialog>
+#include <QFlags>
+#include <QFormLayout>
+#include <QGroupBox>
+#include <QHBoxLayout>
+#include <QIcon>
+#include <QLabel>
+#include <QLatin1String>
+#include <QLineEdit>
+#include <QListWidget>
+#include <QListWidgetItem>
+#include <QModelIndex>
+#include <QPushButton>
+#include <QRadioButton>
+#include <QString>
+#include <QVBoxLayout>
+#include <Qt>
+
+#include "paint_on_template_settings_page.h"
+#include "gui/widgets/settings_page.h"
+#include "settings.h"
+
+class QWidget;
+
+
+namespace OpenOrienteering {
+
+namespace {
+
+struct ColorPresetEntry {
+	const char* name;
+	std::initializer_list<QColor> colors;
+};
+
+constexpr std::array<ColorPresetEntry, 2> color_presets = {
+    //: Paint on template color preset entry name.
+    ColorPresetEntry { QT_TR_NOOP("Traditional"),
```

Comment:
In which "context" is this translation meant to appear for translators?

Reply:
Thanks for making me look into the docs again. The original aim was to provide the right context but I missed the `context` param at `QT_TRANSLATE_NOOP`. Fixed.